File: csharp/triangle/Triangle.cs
using System.Linq;

public static class Triangle
{
    public static bool IsScalene(double side1, double side2, double side3) =>
        IsTriangle(side1, side2, side3) && DistinctSides(side1, side2, side3) == 3;

    public static bool IsIsosceles(double side1, double side2, double side3) =>
        IsTriangle(side1, side2, side3) && DistinctSides(side1, side2, side3) <= 2;

    public static bool IsEquilateral(double side1, double side2, double side3) =>
        IsTriangle(side1, side2, side3) && DistinctSides(side1, side2, side3) == 1;

    private static bool IsTriangle(double side1, double side2, double side3)
    {
        if (side1 <= 0 || side2 <= 0 || side3 <= 0)
            return false;

        return side1 + side2 >= side3 && 
               side1 + side3 >= side2 && 
               side2 + side3 >= side1;
    }

    private static int DistinctSides(double side1, double side2, double side3) =>
        new[] {side1, side2, side3}.Distinct().Count();
}

File: kotlin/triangle/src/main/kotlin/Triangle.kt
class Triangle(a: Double, b: Double, c: Double) {

    constructor(a: Int, b: Int, c: Int): this(a.toDouble(), b.toDouble(), c.toDouble())

    init {
        require(a + b > c)
        require(a + c > b)
        require(b + c > a)
    }

    private val sides = setOf(a, b, c)

    val isEquilateral: Boolean = sides.size == 1
    val isIsosceles: Boolean = sides.size <= 2
    val isScalene: Boolean = sides.size == 3
}
